website/static/js/profile.js

```diff
@@ -323,18 +330,25 @@ BaseViewModel.prototype.cancel = function(data, event) {
 
 BaseViewModel.prototype.submit = function() {
     if (this.hasValidProperty() && this.isValid()) {
-        $osf.putJSON(
-            this.urls.crud,
-            this.serialize()
-        ).done(
-            this.handleSuccess.bind(this)
-        ).done(
-            this.setOriginal.bind(this)
-        ).fail(
-            this.handleError.bind(this)
-        );
-    } else {
-        this.showMessages(true);
+        // Name on Name view is empty
+        if (this.nameFieldEmpty()) {
```

Comment:
I gave the NameViewModel its own submit method and changed BaseViewModel's submit method back to the more general implementation.  Let me know if you think this works.